listenbrainz/db/stats.py

```diff
@@ -0,0 +1,55 @@
+"""This module contains functions to insert and retrieve statistics
+   calculated from Google BigQuery into the database.
+"""
+
+import sqlalchemy
+import ujson
+from listenbrainz import db
+
+
+# XXX(param): think about the names of these stats variables
+# should they be artist_stats and so on instead?
+# Note: names are used in tests and stats/calculate.py also
+
+def insert_user_stats(user_id, artists, recordings, releases, artist_count):
+    # XXX(param): should this name be upsert_user_stats?
+
+    # put all artist stats into one dict which will then be inserted
+    # into the artist column of the stats.user table
+    # XXX(param): This makes the schema of the stats very variable though.
```

Comment:
We have two artist related queries we run for users right now, the results of both of these queries go into the same jsonb field in the db.  The jsonb looks like:
```
{
    "artist_count": 67,
    "all_time": [
        { "name": XXX, "listen_count": XXX}
     ]
}
```

If someone were to add a new artist stat for users, it would go in the same field, so I guess we should document the format of these fields somewhere.